src/openms/source/SYSTEM/File.cpp

```diff
@@ -140,6 +140,73 @@ namespace OpenMS
     return true;
   }
 
+  // https://stackoverflow.com/questions/2536524/copy-directory-using-qt
+  bool File::copyDirRecursively(const QString &from_dir, const QString &to_dir, File::CopyOptions option)
+  {
+    bool overwrite_existing = true;
+    bool skip_existing = false;
+    
+    switch (option)
+    {
+      case CopyOptions::CANCEL: overwrite_existing = false; break;
+      case CopyOptions::SKIP: skip_existing = true; overwrite_existing = false; break;  
+      case CopyOptions::OVERWRITE: break;
+    }   
+ 
+    QDir source_dir(from_dir);
+    QDir target_dir(to_dir);
+
+    QString canonical_source_dir = source_dir.canonicalPath();
+    QString canonical_target_dir = target_dir.canonicalPath();
+   
+    // check canonical path  
+    if (canonical_source_dir == canonical_target_dir)
+    {
+      LOG_ERROR << "Error: Could not copy  " << from_dir.toStdString() << " to " << to_dir.toStdString() << ". Same path given." << std::endl;;  
+      return false;
+    }
+
+    // make directory if not present 
+    if (!target_dir.exists())
+    {
+      target_dir.mkpath(to_dir);
+    }
+  
+    // copy folder recurively
+    QFileInfoList file_list = source_dir.entryInfoList();
+    for (const QFileInfo& entry : file_list)   
+    {
+      if (entry.fileName() == "." || entry.fileName() == "..")
+      {
+        continue;
+      }
+      if (entry.isDir())
+      {
+        if (!copyDirRecursively(entry.filePath(), target_dir.filePath(entry.fileName()), option))
+        {
+          return false;
+        }
+      }
+      else
+      {
+        if (skip_existing && target_dir.exists(entry.fileName()))
```

Comment:
how about:
```
if (target_dir.exists(entry.fileName()))
{
  switch (option)
  {
   case CopyOptions::cancel: 
     return false;
  case CopyOptions::skip:
   continue;
  case CopyOptions::overwrite:
    target_dir.remove(entry.fileName());
  }
}
if (!QFile::copy....
```

Reply:
this way, you can remove the two bools.. less code and a bit cleaner

Reply:
I have not thought about that - thanks!